server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -836,7 +840,22 @@ private VersionValue resolveDocVersion(final Operation op, boolean loadSeqNo) th
             assert incrementIndexVersionLookup(); // used for asserting in tests
             final VersionsAndSeqNoResolver.DocIdAndVersion docIdAndVersion;
             try (Searcher searcher = acquireSearcher("load_version", SearcherScope.INTERNAL)) {
-                docIdAndVersion = VersionsAndSeqNoResolver.loadDocIdAndVersion(searcher.getIndexReader(), op.uid(), loadSeqNo);
+                if (engineConfig.getIndexSettings().getMode() == IndexMode.TIME_SERIES) {
```

Comment:
Does this require its own engine implementation?

Reply:
That seems excessive to me, in particular we may need to repeat this in stateless too then.

Reply:
I see, makes sense. Let's not add another engine implementation.